cumulus/runtime/src/runtime.rs

```diff
@@ -124,7 +124,10 @@ pub mod opaque {
 }
 
 impl_opaque_keys! {
-	pub struct SessionKeys { }
+	pub struct SessionKeys {
+		/// Primarily used for adding the executor authority key into the keystore in the dev mode.
+		pub executor: sp_executor::ExecutorKey,
```

Comment:
Where is this actually used?

Reply:
https://github.com/paritytech/substrate/blob/a5cd3852d144289777dd4b914b6b3aa7d13bcd85/client/service/src/builder.rs#L483-L488 In Substrate, the session keys will be initialized using the dev seed while starting the node, these initialized keys will be stored in the keystone, so that you don't have to insert them into the keystore manually, facilitating the local development.

Reply:
I mean I haven't seen `SessionKeys.executor` field accessed anywhere in the code, but you have just added it. That is why I was confused where is this field actually used. Or does the name not matter at all?

Reply:
No, the name of this field does not matter, what matters is the type of this field that implements `BoundToRuntimeAppPublic` which will tell the authority key type.

Reply:
OK, makes sense, thanks for explanation!

We should probably remove the `subspace` field from `SessionKeys` in primary chain then as we don't use keychain there for anything.